File: backend/spec/system/project_create_spec.rb
require 'rails_helper'

RSpec.describe 'Project management', type: :system do
  include Devise::Test::IntegrationHelpers

  let(:admin) { create(:user, :admin) }

  before do
    sign_in admin
  end

  it 'creates a project with a direct-uploaded attachment', js: true do
    visit '/admin/projects/new'

    fill_in 'project_title', with: 'System Spec Project'
    fill_in 'project_link', with: 'https://example.com/system-spec'
    attach_file 'project_file', Rails.root.join('spec/fixtures/files/test_image.jpg')

    click_button '保存'

    expect(page).to have_current_path('/admin/projects')
    expect(page).to have_content('Project was successfully created.')

    project = Project.order(:created_at).last
    expect(project.title).to eq('System Spec Project')
  end
end

File: e2e/tests/admin/project-create.spec.ts
import { test, expect } from '@playwright/test';
import path from 'path';

const testImage = path.join(__dirname, '..', '..', 'fixtures', 'test_image.jpg');

test.describe('Project management', () => {
  test('creates a project with file upload', async ({ page }) => {
    await page.goto('/admin/projects/new');

    await page.fill('#project_title', 'E2E Test Project');
    await page.fill('#project_link', 'https://example.com/e2e-test');

    await page.setInputFiles('#project_file', testImage);
    await page.click('button:has-text("保存"), input[type="submit"][value="保存"]');

    await expect(page).toHaveURL(/\/admin\/projects$/);
    await expect(page.locator('body')).toContainText('Project was successfully created.');
  });
});
